geode-core/src/main/java/org/apache/geode/internal/cache/Oplog.java

```diff
@@ -5714,6 +5719,7 @@ private void handleEmpty(boolean calledByCompactor) {
       cancelKrf();
       close();
       deleteFiles(getHasDeletes());
+      regionMap.close();
```

Comment:
This line is the only difference than the previous pull request. You moved to handleEmpty() from cleanupAfterCompaction()?

Reply:
Yes. this is the only difference in solution. We have moved regionMap close in handleEmpty() to cover one edge scenario that was not covered with the reverted PR. Also we have added new test case for the scenario. Please find more information's about it in comment section in ticket https://issues.apache.org/jira/browse/GEODE-9903.